### DISCUSSION TITLE
Discharge to grid

### DISCUSSION THREAD

**Discussion by JBayangosB** [wills106/homeassistant-solax-modbus#124] [Q&A]
Hi,

I've recently gotten this integration with with my Gen4 RetroFit X1 inverter, reads all the sensors as far as I can tell accurately and I can change things.

I've been trying to configure the inverter to force discharge for when I move my energy plan to a wholesale price configuration. However, I can't seem to get it to discharge into the grid. Does anyone have experience getting this working? 

These are some settings I have enabled based on looking through other user's comments on force discharge but it doesnt seem to do anything:

![image](https://user-images.githubusercontent.com/16602828/193438471-1819bf5e-03e7-4a4b-9982-06b8c6d16002.png)

Also, despite have `Solax Manual Mode Control` available to me in that screenshot, whenever I change it, it resets back to `Off`. This is whilst having inverter lock state on `unlocked`.

**Comment by JBayangosB** [comment]
I have found this comment on another forum where someone was able to get their solax battery to discharge to the grid by modifying the data being fed to the inverter about house load. 

He's left the code [here](https://forums.moneysavingexpert.com/discussion/comment/79022624/#Comment_79022624)

@wills106 I'm wondering if this could be used with the integration? Happy to modify this locally first to see how I go but would love some guidance on where to begin if possible 🙏

**Reply by wills106** [reply]
They are manipulating the Modbus Power Meter by the looks of it. So you would need a separate RS485 connection connecting to the connection between the Meter and your Inverter.
